src/base.iced

```diff
@@ -178,23 +178,38 @@ class Base
 
   #------
 
-  _v_include_pgp_hash : -> false
-  _v_require_full_pgp_hash : -> false
-  _v_pgp_km_to_hash : () -> @km()
-  _v_pgp_hash_dest : (body) -> body.key
+  _v_include_pgp_details : -> false
+  _v_require_pgp_details : -> false
+  _v_pgp_details_dest : (body) -> body.key
+  _v_pgp_km : () -> @km()
 
   full_pgp_hash : (opts, cb) ->
     if @_full_pgp_hash is undefined
       esc = make_esc cb
-      await @_v_pgp_km_to_hash()?.export_pgp_public {}, esc defer key
+      await @_v_pgp_km()?.export_pgp_public {}, esc defer key
```

Comment:
Can you put in a comment about what would happen if it's a NaCl km?  I.e., that `err` will be `null` but so will `key`.